Make ActiveSupport HashWithIndifferentAccess#to_options and alias for HashWithIndifferentAccess#symbolize_keys

weilandia opened the pull request:
Fixes #34359

### Summary

Prior to 5.2.0 (de9e323, 2cad8d7), `HashWithIndifferentAccess#to_options` acted as
an alias to `HashWithIndifferentAccess#symbolize_keys`. Now, `#to_options`
returns an instance of `HashWithIndifferentAccess` while `#symbolize_keys`
returns and instance of `Hash`.

This pr makes it so `HashWithIndifferentAccess#to_options` acts as an
alias for `HashWithIndifferentAccess#symbolize_keys` once again.

### Other Information

See #34359 

Thanks!

weilandia committed:
Make #to_options an alias for #symbolize_keys

Fixes #34359

Prior to 5.2.0 (2cad8d7), HashWithIndifferentAccess#to_options acted as
an alias to HashWithIndifferentAccess#symbolize_keys. Now, #to_options
returns an instance of HashWithIndifferentAccess while #symbolize_keys
returns and instance of Hash.

This pr makes it so HashWithIndifferentAccess#to_options acts as an
alias for HashWithIndifferentAccess#symbolize_keys once again.

gmcgibbon commented on activesupport/lib/active_support/hash_with_indifferent_access.rb:
```diff
@@ -289,6 +289,7 @@ def deep_stringify_keys; dup end
     undef :symbolize_keys!
     undef :deep_symbolize_keys!
     def symbolize_keys; to_hash.symbolize_keys! end
+    alias_method :to_options,  :symbolize_keys
```
Nitpick: there only needs to be space between `:to_options,` and `:symnolize_keys`.

weilandia replied:
👍

gmcgibbon reviewed (commented):
I think this counts as a bug. Could you please add a changelog entry?

weilandia commented:
@gmcgibbon thanks for the review!  I addressed your comments -- let me know if anything else stands out.

Also, do you think this issue warrants a note in the release notes/upgrade guide for 5.2?
